File: 프로그래머스/1/150370. 개인정보 수집 유효기간/개인정보 수집 유효기간.py
def solution(today, terms, privacies):
    answer = []
    today= today.split('.')
    ty,tm,td = map(int,today)
    today_total = (ty*12*28 + tm*28 + td)

    term = {}
    for t in terms:
        key,value = t.split(' ')
        term[key] = int(value)
        
    for idx,p in enumerate(privacies):
        key,value = p.split(' ')
        jy,jm,jd = map(int,key.split('.'))
        
        if today_total >= (jy*12*28 + jm*28 + jd + term[value]*28):
            answer.append(idx+1)
    
    return answer

File: 프로그래머스/1/150370. 개인정보 수집 유효기간/개인정보 수집 유효기간.js
function solution(today, terms, privacies) {
    var answer = [];
    today = today.split('.');
    const today_total = Number(today[0])*12*28 + Number(today[1])*28 + Number(today[2]);
    
    
    const termObj = {};
    for(let i=0; i<terms.length; i++){
        let term = terms[i].split(' ');
        termObj[term[0]]=Number(term[1]);
    }
    for(let i=0; i<privacies.length; i++){
        let privacy = privacies[i].split(' ');
        let joindate = privacy[0].split('.');
        let total = Number(joindate[0])*12*28 + Number(joindate[1])*28 + Number(joindate[2])+termObj[privacy[1]]*28;
        if(today_total>=total){
            answer.push(i+1);
        }
    }
    return answer;
}